docs/dev/service_levels.md

```diff
@@ -134,3 +135,26 @@ Otherwise, e.g. if a role has multiple workload types declared,
 the conflicts are resolved as follows:
  - `X` vs `unspecified` -> `X`
  - `batch` vs `interactive` -> `batch` - under the assumption that `batch` is safer, because it would not trigger load shedding as eagerly as `interactive`
+
+### Effective service level
+
+Actual values of service level's options may come from different service levels, not only from the one user is assigned with. This can be achived by assigning one role to another.
+
+For instance:
+There are 2 roles: role1 and role2. Role1 is assigned with sl1 (timeout = 2s, workload_type = interactive) and role2 is assigned with sl2 (timeout = 10s, workload_type = batch).
+Then, if we grant role1 to role2, the user with role2 will have 2s timeout (from sl1) and batch workload type (from sl2).
+
+To see detail how the options are merged, check `service/qos/qos_common.cc service_level_options::merge_with()`.
```

Comment:
I now realize that this document has the sentence "In case of timeouts, multiple timeout values are combined by using a minimum of all effective timeouts". This explains in the example above why role2 got sl1's timeout instead of sl2's. I think you should refer to the earlier text about this merging in this document - and not refer to code. You can refer to the code in the text section about this merging, as extra reading.

Reply:
Added a link to section you've mentioned